enhancements/workload-partitioning/wide-availability-workload-partitioning.md

```diff
@@ -0,0 +1,812 @@
+compact(3NC) and HA clusters to specific CPU sets so that I can isolate the
+platform workloads from the application workload due to the high performance and
+determinism required of my applications.
+
+### Goals
+
+- This enhancement describes an approach for configuring OpenShift clusters to
+  run with management workloads on a restricted set of CPUs.
+- Clusters built in this way should pass the same kubernetes and OpenShift
+  conformance and functional end-to-end tests as similar deployments that are
+  not isolating the management workloads.
+- We want to run different workload partitioning on masters and workers.
+- We want to consolidate the current approach and extend PerformanceProfile API
+  to avoid any possible errors when configuring a cluster for workload
+  partitioning.
+- We want to make this enhancement a day 0 supported feature only. We do not
+  support turning it off after the installation is done and the feature is on.
+- We want to make sure that the default behavior of this feature is to allow
+  full use of the CPU set.
+- We want to maintain the ability to configure the partition size after
+  installation, we do not support turning off the partition feature but we do
+  support changing the CPU partition size post day 0. The ability for turning on
+  this feature will be part of the installation phase only.
+
+### Non-Goals
+
+This enhancement expands on the existing [Management Workload
+Partitioning](management-workload-partitioning.md) and as such shares similar
+but slightly different non-goals.
+
+> Note: Items in bold are modifications/additions from the previous enhancement,
+> [Management Workload Partitioning](management-workload-partitioning.md)
+
+- This enhancement is focused on CPU resources. Other compressible resource
+  types may need to be managed in the future, and those are likely to need
+  different approaches.
+- This enhancement does not address non-compressible resource requests, such as
+  for memory.
+- This enhancement does not address ways to disable operators or operands
+  entirely.
+- This enhancement does not address reducing actual utilization, beyond
+  providing a way to have a predictable upper-bounds. There is no expectation
+  that a cluster configured to use a small number of cores for management
+  services would offer exactly the same performance as the default. It must be
+  stable and continue to operate reliably, but may respond more slowly.
+- **This enhancement does not address mixing nodes with pinning and without,
+  this feature will be enabled cluster wide and encapsulate both master and
+  worker pools. If it's not desired then the default behavior will still be
+  turned on but the management workloads will run on the whole CPU set for that
+  desired pool.**
+- **This enhancement assumes that the configuration of a management CPU pool is
+  done as part of installing the cluster. It can be changed after the fact but
```

Comment:
Why do we allow it instead of hard preventing it?

Reply:
The wording here was trying to anticipate a very tenacious customer admin that might decide to apply manual Machine Configs and alter the infra status. We currently support no option to toggle this option. Maybe a better wording would be.

**This enhancement assumes that the configuration of a management CPU pool is
  done as part of installing the cluster since we only support this feature as a day 0 feature. This enhancement does not implement or provide a path to toggle this feature after day 0.**